azurerm/resource_arm_container_group.go

```diff
@@ -531,6 +567,63 @@ func expandContainerEnvironmentVariables(input interface{}) *[]containerinstance
 	return &output
 }
 
+func expandContainerImageRegistryCredentials(d *schema.ResourceData) *[]containerinstance.ImageRegistryCredential {
+	credsRaw := d.Get("image_registry_credential").([]interface{})
+	if len(credsRaw) == 0 {
+		return nil
+	}
+
+	output := make([]containerinstance.ImageRegistryCredential, 0, len(credsRaw))
+
+	for _, c := range credsRaw {
+		credConfig := c.(map[string]interface{})
+
+		server := credConfig["server"].(string)
+		username := credConfig["username"].(string)
+		password := credConfig["password"].(string)
+
+		output = append(output, containerinstance.ImageRegistryCredential{
+			Server:   &server,
+			Password: &password,
+			Username: &username,
+		})
+	}
+
+	return &output
+}
+
+func flattenContainerImageRegistryCredentials(d *schema.ResourceData, credsPtr *[]containerinstance.ImageRegistryCredential) []interface{} {
+	if credsPtr == nil {
+		return nil
+	}
+	configsOld := d.Get("image_registry_credential").([]interface{})
+
+	creds := *credsPtr
+	output := make([]interface{}, len(creds))
+	for _, cred := range creds {
+		credConfig := make(map[string]interface{})
+		if cred.Server != nil {
+			credConfig["server"] = *cred.Server
+		}
+		if cred.Username != nil {
+			credConfig["username"] = *cred.Username
+		}
+
+		for i, configsOld := range configsOld {
```

Comment:
I'm not sure, but i presume the ordering is consistent so you could get the old values this way:

```go
for i, oldIrcConfig := range ircConfigsOld {
...

    if v, ok := d.GetOk(fmt.Sprintf("image_registry_credential.%d.password", i)); ok {
        block["password"] = v.(string)
    }
}
```

Reply:
I'm a little uncertain on this one so was being careful. Does the ordering of elements relate to the ordering of the elements in HCL? So for example if the user update the `password` element in their template but also moved it a line up ahead of `server` would that affect your approach?

Reply:
Yes, the order of the list directly relates to the order of the elements in HCL. So if the user does as you describe, the Apply would be sending a new list anyways, and the subsequent read would be post Update. unless azure returns things in a different order I think it should be fine. You could always write a test to confirm 🙂

Reply:
Cool, makes sense to me. Will also check with the test just to be on the safe side.